static/js/services/report-view-model-generator.js

```diff
@@ -0,0 +1,65 @@
+var _ = require('underscore');
+
+angular.module('inboxServices').factory('ReportViewModelGenerator',
+  function(
+    DB,
+    FormatDataRecord,
+    LineageModelGenerator
+  ) {
+    'ngInject';
+    'use strict';
+
+    var getFields = function(results, values, labelPrefix, depth) {
+      if (depth > 3) {
+        depth = 3;
+      }
+      Object.keys(values).forEach(function(key) {
+        var value = values[key];
+        var label = labelPrefix + '.' + key;
+        if (_.isObject(value)) {
+          results.push({
+            label: label,
+            depth: depth
+          });
+          getFields(results, value, label, depth + 1);
+        } else {
+          results.push({
+            label: label,
+            value: value,
+            depth: depth
+          });
+        }
+      });
+      return results;
+    };
+
+    var getDisplayFields = function(doc) {
+      // calculate fields to display
+      if (!doc.fields) {
+        return [];
+      }
+      var label = 'report.' + doc.form;
```

Comment:
rename `labelPrefix` (I'm twisting my brain around the double `_` thing below and this will help)